kafka/channel/pkg/utils/util.go

```diff
@@ -54,39 +59,32 @@ func GetKafkaConfig(configMap map[string]string) (*KafkaConfig, error) {
 		return nil, fmt.Errorf("missing configuration")
 	}
 
-	config := &KafkaConfig{}
+	config := &KafkaConfig{
+		MaxIdleConns:        DefaultMaxIdleConns,
+		MaxIdleConnsPerHost: DefaultMaxIdleConnsPerHost,
+	}
 
-	if brokers, ok := configMap[BrokerConfigMapKey]; ok {
-		bootstrapServers := strings.Split(brokers, ",")
-		for _, s := range bootstrapServers {
-			if len(s) == 0 {
-				return nil, fmt.Errorf("empty %s value in configuration", BrokerConfigMapKey)
-			}
-		}
-		config.Brokers = bootstrapServers
-	} else {
-		return nil, fmt.Errorf("missing key %s in configuration", BrokerConfigMapKey)
+	var bootstrapServers string
+
+	err := configmap.Parse(configMap,
+		configmap.AsString(BrokerConfigMapKey, &bootstrapServers),
+		configmap.AsInt32(MaxIdleConnectionsKey, &config.MaxIdleConns),
+		configmap.AsInt32(MaxIdleConnectionsPerHostKey, &config.MaxIdleConnsPerHost),
+	)
+	if err != nil {
+		return nil, err
 	}
 
-	if maxConns, ok := configMap["maxIdleConns"]; ok {
-		mc, err := strconv.Atoi(maxConns)
-		if err != nil {
-			config.MaxIdleConns = DefaultMaxIdleConns
-		}
-		config.MaxIdleConns = mc
-	} else {
-		config.MaxIdleConns = DefaultMaxIdleConns
+	if bootstrapServers == "" {
+		return nil, errors.New("missing or empty key bootstrapServers in configuration")
 	}
-	if maxConnsPerHost, ok := configMap["maxIdleConnsPerHost"]; ok {
-		mcph, err := strconv.Atoi(maxConnsPerHost)
-		if err != nil {
-			config.MaxIdleConnsPerHost = DefaultMaxIdleConnsPerHost
+	bootstrapServersSplitted := strings.Split(bootstrapServers, ",")
```

Comment:
a general comment, not for this PR.

For the source @lberk made it array, instead of comma separated list of strings.

Can we do that for the channel too ?

Reply:
I think so and it's even better because there is a ready to use function from the `configmap` parsing module, but this will be breaking